src/backend/cdb/cdbpath.c

```diff
@@ -1102,7 +1102,8 @@ cdbpath_motion_for_join(PlannerInfo *root,
 
 			if (CdbPathLocus_IsReplicated(other->locus))
 			{
-				Assert(root->upd_del_replicated_table > 0);
+				Assert(root->upd_del_replicated_table > 0 ||
+					   (root->parse->hasModifyingCTE && root->parse->commandType == CMD_SELECT));
```

Comment:
Can you remind me, did you test case when we join modifying CTE (under 3-numsegments table) results with 2-numsegments replicated table? We decrease numsegments for `other` locus, but execute it on 3 segments. But data for the 2nd table is located only on 2 segments.

Reply:
As far as I remember, we did that test case. Do you want to see it in tests?

Reply:
Comment added.